### ISSUE TITLE
Pinning Variables on cpu in a multi-gpu enviroment

### ISSUE THREAD

**Issue by ashishkr23** [tflearn/tflearn#459]
Hi, 

Running a small test to check device log placement
tf version 0.11.0rc0
cuda 7.5
According to the tutorial the variables W, b should be placed on the CPU to be shared with all the GPU's but are placed on the gpu:0 itself.

`arg_scope = tf.contrib.framework.arg_scope`
`create_var_op = tf.contrib.framework.python.ops.variables.variable`
`with arg_scope([create_var_op], device='/cpu:0'):`  should force all the variables on the CPU ? or i have understood it incorrectly

Thanks

**Code**

    import tensorflow as tf
    arg_scope = tf.contrib.framework.arg_scope
    create_var_op = tf.contrib.framework.python.ops.variables.variable
    with tf.Graph().as_default(), tf.device('/cpu:0'):
        X = tf.placeholder(tf.float32, [None, 784])
        with tf.device('/gpu:0'):
            with tf.name_scope('tower_0') as scope:
                with arg_scope([create_var_op], device='/cpu:0'):
                    W = tf.get_variable('W', [784, 10])
                    b = tf.get_variable('b', [10])
                    logits = tf.matmul(X, W) + b
        init_op = tf.initialize_all_variables()
        with tf.Session(config=tf.ConfigProto(allow_soft_placement=True,log_device_placement=True)) as sess:
            sess.run(init_op)

**Device Log Placement :**

	Device mapping:
	/job:localhost/replica:0/task:0/gpu:0 -> device: 0, name: GRID K2, pci bus id: 0000:83:00.0
	/job:localhost/replica:0/task:0/gpu:1 -> device: 1, name: GRID K2, pci bus id: 0000:84:00.0
	b/Initializer/random_uniform/min: /job:localhost/replica:0/task:0/gpu:0
	b/Initializer/random_uniform/sub: /job:localhost/replica:0/task:0/gpu:0
	b/Initializer/random_uniform/RandomUniform: /job:localhost/replica:0/task:0/gpu:0
	b/Initializer/random_uniform/mul: /job:localhost/replica:0/task:0/gpu:0
	b/Initializer/random_uniform: /job:localhost/replica:0/task:0/gpu:0
	b: /job:localhost/replica:0/task:0/gpu:0
	b/read: /job:localhost/replica:0/task:0/gpu:0
	b/Assign: /job:localhost/replica:0/task:0/gpu:0
	W/Initializer/random_uniform/min: /job:localhost/replica:0/task:0/gpu:0
	W/Initializer/random_uniform/sub: /job:localhost/replica:0/task:0/gpu:0
	W/Initializer/random_uniform/RandomUniform: /job:localhost/replica:0/task:0/gpu:0
	W/Initializer/random_uniform/mul: /job:localhost/replica:0/task:0/gpu:0
	W/Initializer/random_uniform: /job:localhost/replica:0/task:0/gpu:0
	W: /job:localhost/replica:0/task:0/gpu:0
	W/read: /job:localhost/replica:0/task:0/gpu:0
	W/Assign: /job:localhost/replica:0/task:0/gpu:0
	init: /job:localhost/replica:0/task:0/gpu:0
	tower_0/MatMul: /job:localhost/replica:0/task:0/gpu:0
	tower_0/add: /job:localhost/replica:0/task:0/gpu:0
	b/Initializer/random_uniform/max: /job:localhost/replica:0/task:0/gpu:0
	b/Initializer/random_uniform/shape: /job:localhost/replica:0/task:0/gpu:0
	W/Initializer/random_uniform/max: /job:localhost/replica:0/task:0/gpu:0
	W/Initializer/random_uniform/shape: /job:localhost/replica:0/task:0/gpu:0
	Placeholder: /job:localhost/replica:0/task:0/cpu:0

**Comment by ashishkr23** [comment]
closing this as its more related to tensorflow repo
